packs/vsphere/actions/vm_hw_nic_edit.py

```diff
@@ -0,0 +1,54 @@
+from pyVmomi import vim
+
+from vmwarelib import inventory
+from vmwarelib import checkinputs
+from vmwarelib.actions import BaseAction
+
+
+class VMNicEdit(BaseAction):
+
+    def run(self, vm_id, vm_name, network_adapter, network_name):
+        # check means of finding the VM was provided
+        checkinputs.vm_identifier(vm_id, vm_name)
+        # convert ids to stubs
+        vm = inventory.get_virtualmachine(self.si_content,
+                                          moid=vm_id,
+                                          name=vm_name)
+        network_obj = inventory.get_network(self.si_content,
+                                            name=network_name)
+
+        # find correct NIC
+        for device in vm.config.hardware.device:
+            if isinstance(device, vim.vm.device.VirtualEthernetCard)\
+                    and device.deviceInfo.label == network_adapter:
+                nic = device
+
+        #Different test method due to fact that object
+        #isn't instantiated if not found
+        try:
+            nic
+        except:
+            raise Exception('Unable to find Network Adapter provided')
+
+        # Create object for new Specification
+        new_spec = vim.vm.device.VirtualDeviceSpec()
+        new_spec.operation = vim.vm.device.VirtualDeviceSpec.Operation.edit
+        new_spec.device = nic
+
+        #If network name provided assign new network
+        # Room to expand the following to set additional flags/values
+        if network_name:
```

Comment:
the if network_name line is exactly for that purpose.. none of my actions are intended as absolute.. i hope they will be expanded in the furture. The if network_name was intended to only update what is needed in the spec if someone passes the value.